frontend/packages/noobaa-storage-plugin/src/components/data-consumption-card/data-consumption-card-utils.ts

```diff
@@ -1,148 +1,166 @@
-      x: getMetric(iops, metricType),
-      y: Number(humanizeDecimalBytes(y1).value),
-    };
-    const writes = {
-      name: 'Total Writes',
-      x: getMetric(iops, metricType),
-      y: Number(humanizeDecimalBytes(y2).value),
-    };
-    arr1.push(reads);
-    arr2.push(writes);
-  });
-  return arr1.length > 0 && arr2.length > 0 ? [arr1, arr2] : [];
+export const getQueries: GetQueries = (metric, kpi) => {
+  const queries =
+    DATA_CONSUMPTION_QUERIES[
+      ObjectServiceDashboardQuery[DataConsumersValue[metric] + DataConsumersSortByValue[kpi]]
+    ];
+  const keys = Object.keys(queries);
+  return { queries, keys };
 };
 
-const accountsLogicalUsageChartData: GetBarChartData = (data, metricType) => [
-  data.map((logicalUsage) => ({
-    name: 'Logical Used Capacity',
-    x: getMetric(logicalUsage, metricType),
-    y: Number(humanizeBinaryBytesWithoutB(Number(getValue(logicalUsage))).value),
-  })),
-];
-const providersPhysicalVsLogicalChartData: GetBarChartData = (data) => {
-  const arr1 = [];
-  const arr2 = [];
-  _.forEach(data, (physicalVsLogicalData) => {
-    const y1 = Number(getMetric(physicalVsLogicalData, 'physical_size'));
-    const y2 = Number(getMetric(physicalVsLogicalData, 'logical_size'));
-    const reads = {
-      name: 'Total Physical',
-      x: getMetric(physicalVsLogicalData, 'type'),
-      y: Number(humanizeDecimalBytes(y1).value),
-    };
-    const writes = {
-      name: 'Total Logical',
-      x: getMetric(physicalVsLogicalData, 'type'),
-      y: Number(humanizeDecimalBytes(y2).value),
+export const getChartData: GetChartData = (response, metric, humanize, name) => {
+  const result = _.get(response, 'data.result', []);
+  return result.map((r) => {
+    const x = _.get(r, ['metric', metric], '');
+    const y = parseFloat(_.get(r, 'value[1]'));
+    let val = name;
+    if (!name) val = x; // For Egress having the legends name = providers/accounts name
+    return {
+      name: val,
+      x,
+      y: Number(humanize(y).value),
     };
-    arr1.push(reads);
-    arr2.push(writes);
   });
-  return arr1.length > 0 && arr2.length > 0 ? [arr1, arr2] : [];
 };
 
-const providersEgressChartData: GetBarChartData = (data) => [
-  data.map((egress) => ({
-    name: getMetric(egress, 'type'),
-    x: getMetric(egress, 'type'),
-    y: Number(humanizeBinaryBytesWithoutB(Number(getValue(egress))).value),
-  })),
-];
-
-/* Legends Data Handlers */
-
-const iopsChartLegendData = (data: [BarChartData[], BarChartData[]]) => {
-  const totalReads = data[0].reduce((total: number, iops: BarChartData) => total + iops.y, 0);
-  const totalWrites = data[1].reduce((total: number, iops: BarChartData) => total + iops.y, 0);
-  return [{ name: `Total Reads ${totalReads}` }, { name: `Total Writes ${totalWrites}` }];
+export const getLegendData: GetLegendData = (response, humanize) => {
+  const value = _.get(response, 'data.result[0].value[1]', null);
+  return value ? humanize(Number(value)).string : '';
```

Comment:
No need to convert to number humanize takes care of it.
```suggestion
  return value ? humanize(value).string : '';
```